hadoop-ozone/dist/src/main/smoketest/commonlib.robot

```diff
@@ -35,3 +41,51 @@ Compare files
     ${checksumbefore} =         Execute                    md5sum ${file1} | awk '{print $1}'
     ${checksumafter} =          Execute                    md5sum ${file2} | awk '{print $1}'
                                 Should Be Equal            ${checksumbefore}            ${checksumafter}
+Execute AWSS3APICli
+    [Arguments]       ${command}
+    ${output} =       Execute                    aws s3api --endpoint-url ${ENDPOINT_URL} ${command}
+    [return]          ${output}
+
+Execute AWSS3APICli and checkrc
+    [Arguments]       ${command}                 ${expected_error_code}
+    ${output} =       Execute and checkrc        aws s3api --endpoint-url ${ENDPOINT_URL} ${command}  ${expected_error_code}
+    [return]          ${output}
+
+Execute AWSS3Cli
+    [Arguments]       ${command}
+    ${output} =       Execute                     aws s3 --endpoint-url ${ENDPOINT_URL} ${command}
+    [return]          ${output}
+
+Install aws cli s3 centos
+    Execute            sudo yum install -y awscli
+
+Install aws cli s3 debian
+    Execute            sudo apt-get install -y awscli
+
+Install aws cli
+    ${rc}              ${output} =                 Run And Return Rc And Output           which apt-get
+    Run Keyword if     '${rc}' == '0'              Install aws cli s3 debian
+    ${rc}              ${output} =                 Run And Return Rc And Output           yum --help
+    Run Keyword if     '${rc}' == '0'              Install aws cli s3 centos
+
+Kinit test user
+    ${hostname} =       Execute                    hostname
+    Set Suite Variable  ${TEST_USER}               testuser/${hostname}@EXAMPLE.COM
+    Execute             kinit -k ${TEST_USER} -t /etc/security/keytabs/testuser.keytab
+
+Setup secure credentials
+    Run Keyword         Install aws cli
+    Run Keyword         Kinit test user
+    ${result} =         Execute                    ozone s3 getsecret
+    ${accessKey} =      Get Regexp Matches         ${result}     (?<=awsAccessKey=).*
+    ${secret} =         Get Regexp Matches         ${result}     (?<=awsSecret=).*
+                        Execute                    aws configure set default.s3.signature_version s3v4
+                        Execute                    aws configure set aws_access_key_id ${accessKey[0]}
+                        Execute                    aws configure set aws_secret_access_key ${secret[0]}
+                        Execute                    aws configure set region us-west-1
+
+Setup incorrect credentials for S3
```

Comment:
shall we move this to commonawslib.robot?

Reply:
done